_posts/back-to-basics-messages.md

```diff
@@ -0,0 +1,93 @@
+
+### Events
+
+Events represent the past. "The credit card was charged," "checkout started," or "newsletter published" are valid events.
+
+Events are one-to-many. A publisher publishes an event to zero or more subscribers. A specific publisher publishes events. It's unlikely different publishers will publish the same event.
+
+## Receivers own commands. Publishers own events
+
+One could argue that if we can place orders by phone or online, the "order placed" event could be published by the online ordering process and by the telephone ordering one. True, but that comes with confusion. It's more straightforward if the online process publishes the "online order placed" and the phone one publishes a different event.
+
+If different publishers could publish the same event, subscribers must inspect the incoming message to determine whether they are interested. It's a waste.
+
+Similarly, if different recipients could handle the same command, senders would be required to have logic to determine where to send messages. It needs to be evident who the receiver is.
+
+For those reasons, it's commonly accepted that recipients own commands and publishers own events. Recipients own commands because they own the business logic to process them. Publishers own events because they own the business logic to determine when and why to publish them.
+
+### Why is ownership significant?
+
+We can use ownership to determine who drives changes in the system. For example, a recipient changes a command because the processing logic changes. The changes will percolate down to senders. The opposite applies to events, publishers, and subscribers. For more information on distributed system changes and evolution, look at the [distributed systems evolution article series](https://milestone.topics.it/series/distributed-systems-evolution.html).
+
+## Events cross service boundaries. Commands don't.
+
+Events are, by definition, immutable. They represent something that happened in the past. And the past cannot be changed. Events are good candidates for cross-boundaries communication.
+
+> For an introduction to services, boundaries, and components, please, refer to "[Back to Basics: service boundaries, autonomous components, and coupling](TODO-add-link)."
+
+When crossing a boundary, we're leaving a well-known territory to explore somewhere under someone else's control, a different service. We want to ensure they trust we won't move around the cheese. The second important aspect is that we have no control over someone else lawn. A service cannot tell another service, "Do this for me," it can only broadcast to those interested that something happened.
```

Comment:
```suggestion
When crossing a boundary, we're leaving a well-known territory to explore somewhere under someone else's control, a different service. We want to ensure they trust we won't move around the cheese. The second important aspect is that we have no control over someone else's lawn. A service cannot tell another service, "Do this for me," it can only broadcast to those interested that something happened.
```